### PR TITLE
chore(build): use cross-compilation in the CI builds

### PR DESCRIPTION
**Pull request by plumpy**
(This is what all the other open-source projects have been doing since April. I made a half-hearted effort to get this working then, but obviously I gave up way too quickly.)

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ pull_request_rules:
   - name: Automatically merge on CI success and review
     conditions:
       - base=master
-      - status-success=build (8)
-      - status-success=build (11)
+      - status-success=build (true)
+      - status-success=build (false)
```
Comment (robfletcher) [inline .mergify.yml:6]: I'm not clear what this means

Comment (robfletcher) [reply .mergify.yml:6]: oh, is it because the job name will have `true`/`false` appended from the matrix?

Comment (robfletcher) [reply .mergify.yml:6]: I think I need to go read up about the cross compilation option

Comment (plumpy) [reply .mergify.yml:6]: Yeah, it comes from the matrix. Here's the plugin: https://github.com/nebula-plugins/gradle-java-cross-compile-plugin

Basically it means you can use the Java 11 `javac` but compile against the Java 8 JDK to make sure you don't use new APIs. If you just use `-target 8`, it'll compile to Java 8 bytecodes but successfully compile with methods like `Optional#stream`.

The problem with the jooq is that the `@Generated` annotation moved between Java 8 and Java 11 and using `DETECT_FROM_JDK` uses the Java 11 annotation, so compiling against the Java 8 APIs doesn't work.

Comment (robfletcher) [reply .mergify.yml:6]: So this is good to merge?

Comment (plumpy) [reply .mergify.yml:6]: Yep.

**Pull request merged**